include/mxnet/c_lapack_api.h

```diff
@@ -54,6 +57,40 @@ extern "C" {
 
 inline char loup(char uplo, bool invert) { return invert ? (uplo == 'U' ? 'L' : 'U') : uplo; }
 
+
+/*!
+ * \brief Transpose matrix data in memory
+ * 
+ * Equivalently we can see it as flipping the layout of the matrix
+ * between row-major and column-major.
+ *
+ * \param m number of rows of input matrix a
+ * \param n number of columns of input matrix a
+ * \param b output matrix
+ * \param ldb leading dimension of b
+ * \param a input matrix
+ * \param lda leading dimension of a
+ */
+template <typename xpu, typename DType>
+inline void flip(int m, int n, DType *b, int ldb, DType *a, int lda);
```

Comment:
I was asking for the LAPACK function, was hoping there would be a flag to specify whether any of the inputs needs to be transposed rather than having to do it explicitly ourselves beforehand..